src/wasm/wasm.cpp

```diff
@@ -1155,57 +1155,64 @@ Event* Module::addEvent(Event* curr) {
 
 void Module::addStart(const Name& s) { start = s; }
 
-void Module::removeFunctionType(Name name) {
-  for (size_t i = 0; i < functionTypes.size(); i++) {
-    if (functionTypes[i]->name == name) {
-      functionTypes.erase(functionTypes.begin() + i);
+template<typename Vector, typename Map>
+void removeModuleElement(Vector& v, Map& m, Name name) {
```

Comment:
Would it make sense to implement this in terms of `removeModuleElements`?

Reply:
That might be slower, I worry...

Reply:
For the vector thing the effect would be more or less the same, for the deletion from the map, this can be done in constant time whereas we iterate through all elements in the map in `removeModuleElements`. Given that a lot of usage in the codebase is removing a single element, I guess it'd be better to keep this in the way it is?